src/inversemap.jl

```diff
@@ -0,0 +1,48 @@
+struct InverseMap{T, F, S} <: LinearMap{T}
+    A::F
+    ldiv!::S
+end
+
+"""
+    InverseMap(A; solver = ldiv!)
+
+Lazy inverse of `A` such that `InverseMap(A) * x` is the same as `A \\ x`. Letting an
+`InverseMap` act on a vector thus requires solving a linear system.
+
+A solver function can be passed with the `solver` keyword argument. The solver should
+be of the form `f(y, A, x)` where `A` is the wrapped map, `x` the right hand side, and
+`y` a preallocated output vector in which the result should be stored. The default solver
+is `LinearAlgebra.ldiv!`.
+
+Note that `A` must be compatible with the solver function. `A` can, for example`, be a
+factorization of a matrix, or another another `LinearMap` (in combination with an
+iterative solver such as conjugate gradient).
+
+# Examples
+```julia
+julia> using LinearMaps, LinearAlgebra
+
+julia> A = rand(2, 2); b = rand(2);
+
+julia> InverseMap(lu(A)) * b
+2-element Vector{Float64}:
+  1.0531895201271027
+ -0.4718540250893251
+
+julia> A \\ b
+2-element Vector{Float64}:
+  1.0531895201271027
+ -0.4718540250893251
+```
+"""
+function InverseMap(A::F; solver::S=LinearAlgebra.ldiv!) where {F, S}
+    T = eltype(A)
+    InverseMap{T,F,S}(A, solver)
+end
+
+Base.size(imap::InverseMap) = size(imap.A)
+
+function LinearMaps._unsafe_mul!(y::AbstractVecOrMat, imap::InverseMap, x::AbstractVector)
```

Comment:
This caused an ambiguity error actually.

Reply:
Yea, that was a bad idea as I realized once I tried with the other map types. Still, I think we will want to have another copy (sigh!) of the method with an `x::AbstractMatrix` signature, to allow the solver handle matrices. Otherwise, `LinearMaps.jl` will apply the solve columnwise, which may miss matrix optimizations of the solver.